src/compiler/moduleNameResolver.ts

```diff
@@ -188,7 +188,10 @@ namespace ts {
 
         let resolvedTypeReferenceDirective: ResolvedTypeReferenceDirective | undefined;
         if (resolved) {
-            resolved = realpath(resolved, host, traceEnabled);
+            if (!options.preserveSymlinks) {
+                resolved = realpath(resolved, host, traceEnabled);

```

Comment:
The first action taken by `realpath` here is to test `if (!host.realpath) return path;`
I can change the name of the local function but wouldn't changing the name of the host method break existing hosts?